nipype/algorithms/compcor.py

```diff
@@ -0,0 +1,81 @@
+from ..interfaces.base import (BaseInterfaceInputSpec, TraitedSpec,
+                               BaseInterface, traits, File)
+import nibabel as nb
+import numpy as np
+from scipy import linalg, stats
+import os
+
+from nipype.pipeline.engine import Workflow
+
+class CompCoreInputSpec(BaseInterfaceInputSpec):
+    realigned_file = File(exists=True, mandatory=True, desc='already realigned brain image (4D)')
+    mask_file = File(exists=True, mandatory=True, desc='mask file that determines ROI (3D)')
+    num_components = traits.Int(6, usedefault=True) # 6 for BOLD, 4 for ASL
+    # additional_regressors??
+
+class CompCoreOutputSpec(TraitedSpec):
+    components_file = File(desc='text file containing the noise components', exists=True)
+
+class CompCore(BaseInterface):
```

Comment:
Yes it's a pun :) I'll change it.

re doi's, is this what you're referring to? https://github.com/nipy/nipype/issues/1464